### DISCUSSION TITLE
Optional programmatic default blockName field

### DISCUSSION THREAD

**Discussion by damnsamn** [payloadcms/payload#4648] [Feature Requests & Ideas]
It would be great to be able to define a programmatic default `blockName` pattern, usage similar to `admin.components.RowLabel`.

My thinking is that a block's `blockName` would default to the generated output of a given function while the input is empty - automatically updating each time the block's data updates. `blockName` could still be overridden in the UI, which would break the link between it and the generator function. When cleared the input is cleared, it would default back to the provided function output.

Example:

```
const MyBlock: Block = {
  slug: "myBlock",
  defaultBlockName: ({data}) => data.name,
  fields: [
    {
      type: "text",
      name: "name",
      required: true,
    }
  ]
} 
```
 
<img width="316" alt="image" src="https://github.com/payloadcms/payload/assets/17864782/f2e9abef-48cf-4751-9aad-74794e9caf1d">

**Comment by damnsamn** [comment]
Example use case:

Here is a Navigation Menu builder which utilises Blocks to differentiate between navigable menu items, and non-navigable menu item parents (2nd-level navigable only)

In this case, the user will have to self-label each block in the list by copying-and-pasting from the `label` field.

<img width="1030" alt="image" src="https://github.com/payloadcms/payload/assets/17864782/9a38fc36-3c18-4bcd-8d94-a74ec9331f83">

In an ideal scenario, we could define these blocks as being automatically labelled by its `label` field by default, which the user could optionally override. This would avoid the "Untitled" default which forces them to duplicate effort in naming each menu item

**Comment by cbrualdi** [comment]
No one interested in this?

I think this feature is crucial for the usability and clarity of the CMS. Blocks are often used to create complex structures, and the information identifying a block is often already contained within the internal fields. We also have several use cases where it is useful to generate a blockName from multiple internal fields, but people either don't use it or are bored with re-entering the same information.

In my opinion, this change would add a lot of value for the end user.

What do you think?

**Comment by lotarbo** [comment]
It's super useful, for example in sanity you can do that. https://www.sanity.io/docs/previews-list-views

**Comment by damnsamn** [comment]
This still gets regularly raised as a bug, both internally and from clients. Having to explain "Payload doesn't support this" fairly often at this point

**Comment by GermanJablo** [comment]
This is supported, it just wasn't well documented. Please see the changes in the documentation from: https://github.com/payloadcms/payload/pull/13782

**Reply by damnsamn** [reply]
The other comment is obviously unhelpful, but I do agree that this is not really a good solution to this.

Every single blocks field across every site I've worked on in Payload has looked like this:
<img width="859" height="848" alt="image" src="https://github.com/user-attachments/assets/53a86e62-3352-4dfb-8b1b-4c7b27aa9f60" />

The fact that the onus is on the content editor to update every `blockName` manually, and manually keep it in sync with the content of the block itself - with no first-class way for the developer to programatically set or default that value based on the block's own content - means that it **never** gets set, and the Payload CMS experience overall is degraded.

**Comment by robclancy** [comment]
That isn't even a good solution to this. I don't see why we can't just set the damn blockName. I can't go a week without finding some weird decision in payload.
